x-pack/plugins/snapshot_restore/public/plugin.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { PLUGIN } from '../common/constants';
+import { CLIENT_BASE_PATH, renderReact } from './app';
+import template from './index.html';
+import { AppCore, AppPlugins, Core, Plugins } from './shim';
+
+const REACT_ROOT_ID = 'snapshotRestoreReactRoot';
+
+export class Plugin {
+  public start(core: Core, plugins: Plugins): void {
+    const { i18n, routing, http, chrome, notification } = core;
```

Comment:
`routing` seems like maybe not the best name for this function, since it has unmount logic, which seems unrelated to routing, more like rendering to me.

Reply:
good point, I'll move unmounting logic out of shim, and keep `routing` for just routing